File: Server/Server/Demo/logic/MainMsgHandle.cs
using System;
using YT;

namespace YT
{
    public partial class MsgHandler
    {

        //获取个人简介内容
        public static void MsgGetPlayerIntroduction(Connection c, MsgBase msgBase) {
            MsgGetPlayerIntroduction msg = (MsgGetPlayerIntroduction)msgBase;
            Player player = c.player;
            if (player == null) return;
            //获取text
            msg.palyerIntroduction = player.data.palyerIntroduction;
            player.Send(msg);
        }

        //保存个人简介内容
        public static void MsgSavePlayerIntroduction(Connection c, MsgBase msgBase) {
            MsgSavePlayerIntroduction msg = (MsgSavePlayerIntroduction)msgBase;
            Player player = c.player;
            if (player == null) return;
            //获取头像
            player.data.palyerIntroduction = msg.palyerIntroduction;
            //保存数据
            DBManager.UpdatePlayerData(c.player.id, c.player.data);
            player.Send(msg);
        }

        //获取头像
        public static void MsgGetHeadPhoto(Connection c, MsgBase msgBase) {
            MsgGetHeadPhoto msg = (MsgGetHeadPhoto)msgBase;
            Player player = c.player;
            if (player == null) return;
            //获取头像
            msg.headPhoto = player.data.headPhoto;
            player.Send(msg);
        }

        //保存头像
        public static void MsgSaveHeadPhoto(Connection c, MsgBase msgBase) {
            MsgSaveHeadPhoto msg = (MsgSaveHeadPhoto)msgBase;
            Player player = c.player;
            if (player == null) return;
            //获取头像
            player.data.headPhoto = msg.headPhoto;
            //保存数据
            DBManager.UpdatePlayerData(c.player.id, c.player.data);
            player.Send(msg);
        }

        //发送消息到世界
        public static void MsgSendMessageToWord(Connection c, MsgBase msgBase) {
            MsgSendMessageToWord msg = (MsgSendMessageToWord)msgBase;

            Player player = c.player;
            //返回协议
            msg.result = 0;
            PlayerManager.Broadcast(msg);


        }

        ////发送消息到好友
        public static void MsgSendMessageToFriend(Connection c, MsgBase msgBase) {
            MsgSendMessageToFriend msg = (MsgSendMessageToFriend)msgBase;


        }

        //获取好友列表
        public static void MsgGetFriendList(Connection c, MsgBase msgBase) {
            MsgGetFriendList msg = (MsgGetFriendList)msgBase;
            Player player = c.player;
            if (player == null) return;
            //获取FriendList
            msg.friendIdList = DBManager.GetFriendList(c.player.id);
            msg.result = 0;
            player.Send(msg);
        }
        //添加好友
        public static void MsgAddFriend(Connection c, MsgBase msgBase) {
            MsgAddFriend msg = (MsgAddFriend)msgBase;
            Player player = c.player;
            if (player == null) return;

            if (DBManager.IsAccountExist(msg.friendId)) {
                msg.result = 1;
                player.Send(msg);
                return;
            }
            if (!DBManager.IsFriendExist(msg.id, msg.friendId)) {
                msg.result = 3;
                player.Send(msg);
                return;
            }
            if (PlayerManager.GetPlayer(msg.friendId) == null) {
                msg.result = 2;
                player.Send(msg);
                return;
            }
            msg.result = 0;
            player.Send(msg);
            PlayerManager.GetPlayer(msg.friendId).Send(msg);
        }

        //删除好友
        public static void MsgDeleteFriend(Connection c, MsgBase msgBase) {
            MsgDeleteFriend msg = (MsgDeleteFriend)msgBase;
            Player player = c.player;
            if (player == null) return;
            if (DBManager.DeleteFriend(player.id, msg.friendId)) {
                msg.result = 0;
                player.Send(msg);
                if (PlayerManager.GetPlayer(msg.friendId) != null) {
                    PlayerManager.GetPlayer(msg.friendId).Send(msg);
                }

            }
            else {
                msg.result = 1;
                player.Send(msg);
            }


        }
        //同意添加好友
        public static void MsgAcceptAddFriend(Connection c, MsgBase msgBase) {
            MsgAcceptAddFriend msg = (MsgAcceptAddFriend)msgBase;
            Player player = c.player;
            if (player == null) return;

            if (DBManager.AddFriend(msg.id, msg.friendId)) {
                msg.result = 0;
            }
            else {
                msg.result = 1;
            }
            player.Send(msg);
            PlayerManager.GetPlayer(msg.friendId).Send(msg);
        }
    }
}

File: Server-Go/Demo/MsgHandler/MainMsgHandle.go
package MsgHandler

import (
	"ReaminHistory/Demo/DB"
	"ReaminHistory/Demo/Player/PlayerManager"
	"ReaminHistory/Demo/Proto"
	"ReaminHistory/YT/Net"
	"fmt"
	"github.com/golang/protobuf/proto"
)

//获取个人简介内容
func MsgGetPlayerIntroduction(c *Net.Connection, bytes []byte) {
	msg := &Proto.MsgGetPlayerIntroduction{}
	err := proto.Unmarshal(bytes, msg)
	if err != nil {
		fmt.Println("解析失败")
		return
	}
	if player, ok := PlayerManager.Players[c.Conv]; ok {
		msg.PalyerIntroduction = player.Data.PlayerIntroduction
		player.Send(msg)
	} else {
		return
	}
}

//保存个人简介内容
func MsgSavePlayerIntroduction(c *Net.Connection, bytes []byte) {
	msg := &Proto.MsgSavePlayerIntroduction{}
	err := proto.Unmarshal(bytes, msg)
	if err != nil {
		fmt.Println("解析失败")
		return
	}
	if player, ok := PlayerManager.Players[c.Conv]; ok {
		player.Data.PlayerIntroduction = msg.PalyerIntroduction
		DB.UpdatePlayerData(player.Id, player.Data)
		player.Send(msg)
	} else {
		fmt.Println("获取失败")
		return
	}
}

//获取头像
func MsgGetHeadPhoto(c *Net.Connection, bytes []byte) {
	msg := &Proto.MsgGetHeadPhoto{}
	err := proto.Unmarshal(bytes, msg)
	if err != nil {
		fmt.Println("解析失败")
		return
	}
	if player, ok := PlayerManager.Players[c.Conv]; ok {
		msg.HeadPhoto = player.Data.HeadPhoto
		player.Send(msg)
	} else {
		fmt.Println("获取失败")
		return
	}
}

//保存头像
func MsgSaveHeadPhoto(c *Net.Connection, bytes []byte) {
	msg := &Proto.MsgSaveHeadPhoto{}
	err := proto.Unmarshal(bytes, msg)
	if err != nil {
		fmt.Println("解析失败")
		return
	}
	if player, ok := PlayerManager.Players[c.Conv]; ok {
		player.Data.HeadPhoto = msg.HeadPhoto
		DB.UpdatePlayerData(player.Id, player.Data)
		player.Send(msg)
	} else {
		return
	}
}

//发送消息到世界
func MsgSendMessageToWord(c *Net.Connection, bytes []byte) {
	msg := &Proto.MsgSendMessageToWord{}
	err := proto.Unmarshal(bytes, msg)
	if err != nil {
		fmt.Println("解析失败")
		return
	}
	if _, ok := PlayerManager.Players[c.Conv]; ok {
		msg.Result = 0
		PlayerManager.Broadcast(msg)
	} else {
		return
	}
}

//发送消息到好友
func MsgSendMessageToFriend(c *Net.Connection, bytes []byte) {
	msg := &Proto.MsgSendMessageToFriend{}
	err := proto.Unmarshal(bytes, msg)
	if err != nil {
		fmt.Println("解析失败")
		return
	}
	//TODO
}

//获取好友列表
func MsgGetFriendList(c *Net.Connection, bytes []byte) {
	msg := &Proto.MsgGetFriendList{}
	err := proto.Unmarshal(bytes, msg)
	if err != nil {
		fmt.Println("解析失败")
		return
	}
	if player, ok := PlayerManager.Players[c.Conv]; ok {
		msg.FriendList = DB.GetFriendList(player.Id)
		msg.Result = 0
		player.Send(msg)
	} else {
		return
	}
}

//添加好友
func MsgAddFriend(c *Net.Connection, bytes []byte) {
	msg := &Proto.MsgAddFriend{}
	err := proto.Unmarshal(bytes, msg)
	if err != nil {
		fmt.Println("解析失败")
		return
	}
	if player, ok := PlayerManager.Players[c.Conv]; ok {
		if DB.IsAccountExist(msg.FriendId) {
			msg.Result = 1
			player.Send(msg)
			return
		}
		if DB.IsFriendExist(msg.Id, msg.FriendId) {
			msg.Result = 2
			player.Send(msg)
			return
		}
		if PlayerManager.GetPlayer(msg.FriendId) == nil {
			msg.Result = 3
			player.Send(msg)
			return
		}
		msg.Result = 0
		player.Send(msg)
		PlayerManager.GetPlayer(msg.FriendId).Send(msg)
	} else {
		return
	}
}

//删除好友
func MsgDeleteFriend(c *Net.Connection, bytes []byte) {
	msg := &Proto.MsgDeleteFriend{}
	err := proto.Unmarshal(bytes, msg)
	if err != nil {
		fmt.Println("解析失败")
		return
	}
	if player, ok := PlayerManager.Players[c.Conv]; ok {
		if DB.DeleteFriend(msg.Id, msg.FriendId) {
			msg.Result = 0
			player.Send(msg)
			if PlayerManager.GetPlayer(msg.FriendId) != nil {
				PlayerManager.GetPlayer(msg.FriendId).Send(msg)
			}
		} else {
			msg.Result = 1
			player.Send(msg)
		}
	} else {
		return
	}
}

//同意添加好友
func MsgAcceptAddFriend(c *Net.Connection, bytes []byte) {
	msg := &Proto.MsgAcceptAddFriend{}
	err := proto.Unmarshal(bytes, msg)
	if err != nil {
		fmt.Println("解析失败")
		return
	}
	if player, ok := PlayerManager.Players[c.Conv]; ok {
		if DB.AddFriend(msg.Id, msg.FriendId) {
			msg.Result = 0
		} else {
			msg.Result = 1
		}
		player.Send(msg)
		PlayerManager.GetPlayer(msg.FriendId).Send(msg)
	} else {
		return
	}
}
